### ISSUE TITLE
HTML in textarea

### ISSUE THREAD

**Issue by katlandreth** [lonekorean/highlight-within-textarea#4]
Thanks for the great plugin, it's perfect for my app!  This is probably a pretty newb question, so apologies in advance.

My textarea accepts HTML markup, so the duplicated div behind the textarea was actually rendering images instead of the markup for those images.

Right now I only need to worry about `figure`, `figcaption`, and  `img`  tags, so I'm escaping that HTML below line 133 in highlight-within-textarea.js

``` javascript
//this escapes figure and img tags in the input
input = input.replace(/&/, '&amp;')
    .replace(/<figure>/, '&lt;figure&rt;')
    .replace(/<figcaption>/, '&lt;figcaption&rt;')
    .replace(/<img/, '&lt;img');
```

This works fine for now, but I actually allow more HTML than that in this textarea and writing replace() for each tag will get cumbersome. Is there a way I can escape all of the HTML _except_ the `<mark>` tags?

**Comment by katlandreth** [comment]
Found a better way. Escape the HTML characters (stabby brackets etc), then re-replace the `mark` tag.  Not sure it's the best way, but it works for what I need. 

``` javascript
input = input.replace(/&/, '&amp;')
    .replace(/"/g, '&quot;')
    .replace(/</g, '&lt;')
    .replace(/>/g, '&gt;')
    .replace(/&lt;mark&gt;/g,'<mark>')
    .replace(/&lt;\/mark&gt;/g,'</mark>');
```

**Comment by lonekorean** [comment]
Thanks for the bug report. You're totally right, it injects the markup into the page, unencoded. This is a potential XSS concern too, since you can inject the &lt;script&gt; tag. That's no good.

**Comment by lonekorean** [comment]
Published version 1.0.3. HTML input should work fine now and the XSS vulnerability removed.

Thanks again for spotting this!
